compiler-plugin/src/main/scala/scalapb/compiler/ProtobufGenerator.scala

```diff
@@ -1048,14 +1072,23 @@ class ProtobufGenerator(
   def generateMessageCompanionMatcher(methodName: String, messageNumbers: Seq[(Descriptor, Int)])(
       fp: FunctionalPrinter
   ): FunctionalPrinter = {
-    val signature =
+    val signature = if (params.scala3Sources) {
+      s"def $methodName(__number: _root_.scala.Int): _root_.scalapb.GeneratedMessageCompanion[?] = "
```

Comment:
Since there are a bunch of `if..else` s around the wildcard, I suggest we add a `WildcardType` constant to `Constants`. Then here and below just interpolate using `WildcardType`.